language-plutus-core/src/Language/PlutusCore/TypeSynthesis.hs

```diff
@@ -54,7 +59,7 @@ bsRel = builtinRel TyByteString
 -- | Create a dummy 'TyName'
 newTyName :: (MonadQuote m) => Kind () -> m (TyNameWithKind ())
 newTyName k = do
-    u <- nameUnique . unTyName <$> liftQuote (freshTyName () "a")
+    u <- liftQuote freshUnique
```

Comment:
Unrelated, but perhaps we should generalize functions like `freshUnique` to work for any `MonadQuote m`?